leader-election-impl/src/main/java/com/palantir/paxos/SingleLeaderPinger.java

```diff
@@ -137,7 +181,7 @@ private LeaderPingResult actuallyPingLeaderWithUuid(
             Future<Map.Entry<LeaderPingerContext<PingableLeader>, PingResult>> pingFuture =
                     multiplexingCompletionService.poll(leaderPingResponseWait.toMillis(), TimeUnit.MILLISECONDS);
             return getLeaderPingResult(
-                    uuid, pingFuture, timeLockVersion, greeningNodeShouldBecomeLeaderRateLimiter::tryAcquire);
+                    uuid, pingFuture, timeLockVersion, greenNodeLeadershipPrioritiser::shouldGreeningNodeBecomeLeader);
```

Comment:
side node: It seems to me like this logic is in the wrong place - it's a bit strange for `SingleLeaderPinger` to effectively lie about the ping result (hiding the older version) based on the leadership prioritiser logic.

IMHO `SingleLeaderPinger` should return the correct information about the current leader, and `PaxosLeaderElectionService` (or some extracted class) should be responsible for deciding what to do with that information.

However, it looks like being quite an involved (and maybe risky) refactor, so I definitely don't think it belongs in this PR.